### ISSUE TITLE
Simplified query params issues 

### ISSUE THREAD

**Issue by tdipisa** [geosolutions-it/MapStore2#8644]
## Description

Issues found during functional testing in DEV

*Expected Result*

https://github.com/geosolutions-it/MapStore2/wiki/%5BProposal%5D-MapStore-query-parameters-simplification

*Current Result*

`?featureinfo=42.212,-108.369

working

`?addLayers=gs:us_states`

working

`?addLayers=gs:us_states&bbox=-117,43,-101,49` 

working

`?addLayers=gs:us_states&center=-106.458,45.699&zoom=5&mapInfo=gs:us_states&mapInfoFilter=STATE_FIPS=30`

Identify not triggered

`?addLayers=gs:us_states&bbox=-117,43,-101,49&mapInfo=gs:us_states&mapInfoFilter=STATE_FIPS=30`

not working, identify not showing up (without _bbox_ works)

`?addLayers=gs:spearfish_roads,gs:spearfish_bugsites;gs_stable_csw,gs:spearfish_archsites;gs_stable_csw,gs:spearfish_restricted&mapInfo=gs:spearfish_restricted&mapInfoFilter=cat=3`

working

`?addLayers=gs:spearfish_roads,gs:spearfish_bugsites;gs_stable_csw,gs:spearfish_archsites;gs_stable_csw,gs:spearfish_restricted&mapInfo=gs:spearfish_restricted&mapInfoFilter=cat=3&bbox=-103.882,44.374,-103.618,44.501`

Identify not working

`?addLayers=gs:spearfish_roads,gs:spearfish_bugsites;gs_stable_csw,gs:spearfish_archsites;gs_stable_csw,gs:spearfish_restricted&mapInfo=gs:spearfish_restricted&mapInfoFilter=cat=3&layerFilters=cat='4',str1='Beetle site',str1='Bob Miller',cat='3'`

_layerFilters_ is not processed as expected if multiple filters are present (one for each layer in the list) it works only if one filter is defined

`?addLayers=gs:spearfish_roads,gs:spearfish_bugsites;gs_stable_csw,gs:spearfish_archsites;gs_stable_csw,gs:spearfish_restricted&mapInfo=gs:spearfish_restricted&mapInfoFilter=cat=3&layerFilters=cat='4'&background=gs:spearfish_dem;gs_stable_wms` 

not working as expected _background_ only works if the source is specified while it should work also without specifying it with the default source

`?addLayers=gs:spearfish_roads,gs:us_states,gs:spearfish_bugsites;gs_stable_csw,gs:spearfish_archsites;gs_stable_csw,gs:spearfish_restricted&layerFilters=cat='4'&background=gs:us_states;gs_stable_wms&actions=[{"type":"SEARCH:SEARCH_WITH_FILTER","cql_filter":"STATE_FIPS=34","layer":"gs:us_states"}] `

layer _gs:us_states_ seems not queryable (but it should be) and a MS notofication appears. Why?

- [x] Not browser related

<details><summary> <b>Browser info</b> </summary>

(use this site: <a href="https://www.whatsmybrowser.org/">https://www.whatsmybrowser.org/</a> for non expert users)

| Browser Affected | Version |
|---|---|
|Internet Explorer| |
|Edge| |
|Chrome| |
|Firefox| |
|Safari| |
</details>

**Comment by alex-fko** [comment]
@tdipisa 
Issue with `mapInfo` parameter not triggering identify panel to open should be resolved now, there was a race condition that was making it to fail in some cases.


> layerFilters is not processed as expected if multiple filters are present (one for each layer in the list) it works only if one filter is defined

It expects a  semicolon ";" delimiter, please refer to this comment https://github.com/geosolutions-it/MapStore2/pull/8617#pullrequestreview-1124819050
Documentation is also reflects that in this case semicolon is a delimiter (https://mapstore.readthedocs.io/en/latest/developer-guide/map-query-parameters/#addlayers formatting issue is fixed by the connected PR)

> not working as expected background only works if the source is specified while it should work also without specifying it with the default source
default service for backgrounds is the 'default_map_backgrounds' containing all the backgrounds layers configured in localConfig. Should this behavior change to also use default catalog service?

> layer gs:us_states seems not queryable (but it should be) and a MS notofication appears. Why?

This is expected behavior, `SEARCH:SEARCH_LAYER_WITH_FILTER` is an action triggering search flow instantly. There is a different action called `SEARCH:SCHEDULE_SEARCH_LAYER_WITH_FILTER` used by a `mapInfo` parameter handler that is dispatched if it finds that `mapInfo` and `addLayers` have the same layer name. It delays the actual `SEARCH:SEARCH_LAYER_WITH_FILTER` appearance until layer is added to the map. So this approach will work:
1. Add `initialActionsWhiteList` into localConfig:
```
"initialActionsWhiteList": ["SEARCH:SCHEDULE_SEARCH_LAYER_WITH_FILTER"]
```
2. Query newly added action instead of original one that search instantly:
```
?addLayers=gs:spearfish_roads,gs:us_states,gs:spearfish_bugsites;gs_stable_csw,gs:spearfish_archsites;gs_stable_csw,gs:spearfish_restricted&layerFilters=cat='4'&actions=[{"type":"SEARCH:SCHEDULE_SEARCH_LAYER_WITH_FILTER","cql_filter":"STATE_FIPS=34","layer":"gs:us_states"}]
```

**Comment by tdipisa** [comment]
@alex-fko thank you for the update.

**Comment by offtherailz** [comment]
@ElenaGallo, could you please test this on  DEV  ? Thank you

**Comment by ElenaGallo** [comment]
Test passed on DEV, @alex-fko please backport. Thanks

**Comment by offtherailz** [comment]
Everything looks good to me. I only noticed this delay in applying the changes, but I think it is a known issue @tdipisa 

As well as I open the map with parameters, first of all the original map is load, then the changes are applied: See animation here:

https://user-images.githubusercontent.com/1279510/193771439-73918450-45e9-40d7-9493-a8f4a4f255a0.mp4

**Comment by tdipisa** [comment]
@offtherailz 

>Everything looks good to me. I only noticed this delay in applying the changes, but I think it is a known issue @tdipisa 

yes, that's the desired behavior otherwise it is not possible to combine different actions together acting on the same layer (eg. addlayer and identify on the same layer)

**Comment by offtherailz** [comment]
So we can backport ;)

**Comment by tdipisa** [comment]
Wait @offtherailz 

@alex-fko I've added checkboxes above for points that are now working after my last test. The following still don't work anyway:

> ?addLayers=gs:spearfish_roads,gs:spearfish_bugsites;gs_stable_csw,gs:spearfish_archsites;gs_stable_csw,gs:spearfish_restricted&mapInfo=gs:spearfish_restricted&mapInfoFilter=cat=3&layerFilters=cat='4'&background=gs:spearfish_dem
 
![image](https://user-images.githubusercontent.com/1280027/193778166-24c63fb5-a67a-40fe-84dd-7ce278d7e5b8.png)

>default service for backgrounds is the 'default_map_backgrounds' containing all the backgrounds layers configured in localConfig. Should this behavior change to also use default catalog service?

Yes, the default catalog source should be used as per original proposal.

Then:

>?addLayers=gs:spearfish_roads,gs:us_states,gs:spearfish_bugsites;gs_stable_csw,gs:spearfish_archsites;gs_stable_csw,gs:spearfish_restricted&layerFilters=cat='4'&actions=[{"type":"SEARCH:SCHEDULE_SEARCH_LAYER_WITH_FILTER","cql_filter":"STATE_FIPS=34","layer":"gs:us_states"}]

This is working but it would be good to effectively include the new action in the default ones in localConfig and also documented [here](https://mapstore.readthedocs.io/en/latest/developer-guide/map-query-parameters/#available-actions).

**Comment by ElenaGallo** [comment]
Test passed on DEV, @alex-fko please backport to 2022.02.xx. Thanks
